core/client/fs/src/main/java/alluxio/client/RetryHandlingMetaMasterClient.java

```diff
@@ -45,9 +46,10 @@
    * Creates a new meta master client.
    *
    * @param conf master client configuration
+   * @param alluxioConf Alluxio's configuration
    */
-  public RetryHandlingMetaMasterClient(MasterClientConfig conf) {
-    super(conf);
+  public RetryHandlingMetaMasterClient(MasterClientConfig conf, AlluxioConfiguration alluxioConf) {
```

Comment:
I would say no because the configuration is only needed to create a MasterInquireClient. Even after the `MasterInquireClient` is created there's no need to store the configuration in the MasterClientConfig because the `MasterInquireClient` doesn't hold a reference to a `MasterClientConfig`.

In fact it's the other way around - the `MasterClientConfig` already typically holds a reference to the configuration.

Reply:
Well now that I actually wrote out the class diagram I guess it would make sense to store the configuration? But then because it also has a subject field should we just make it a `ClientContext` instead?

Reply:
I'm fine either way, as long as `MasterClientConfig` includes all the config for the master client

Reply:
how do you feel about removing `defaults` and forcing the user to specifically set the configuration, subject, and master inquire client? I don't care but I feel like the name `defaults()` is kind of misleading since we're passing the configuration arguments?

Reply:
it's really just a lazy version of the builder pattern where the `MasterClientConfig` class is its own builder.

Using a builder style scales better than constructor parameters and makes it easy to see at the call site what each parameter means. How do you like this?
```java
  MasterClientConfig.newBuilder(conf)
    .withSubject(subject)
    .withMasterInquireClient(masterInquireClient)
    .build()
```

Reply:
I like this better. I'll address this once I get some more fixes out